packages/create-plugin/templates/backend/src/datasource.ts

```diff
@@ -9,6 +9,18 @@ export class DataSource extends DataSourceWithBackend<MyQuery, MyDataSourceOptio
   }
 
   getDefaultQuery(_: CoreApp): Partial<MyQuery> {
-    return DEFAULT_QUERY
+    return DEFAULT_QUERY;
+  }
+
+  applyTemplateVariables(query: MyQuery, scopedVars: ScopedVars): Record<string, any> {
+    return {
+      ...query,
+      queryText: getTemplateSrv().replace(query.queryText, scopedVars),
```

Comment:
Regarding scoped variables...currently the ds template for `plugin.json` doesn't specify any [query options](https://grafana.com/developers/plugin-tools/reference-plugin-json#queryoptions). Should we skip passing the scopredVars variable to `getTemplateSrv().replace`?